src/highdicom/seg/sop.py

```diff
@@ -541,15 +545,19 @@ def __init__(
             self.DimensionIndexSequence.get_index_values(plane_positions)
 
         if self._coordinate_system == CoordinateSystemNames.SLIDE:
-            self.ImageOrientationSlide = \
+            self.ImageOrientationSlide = deepcopy(
                 plane_orientation[0].ImageOrientationSlide
-            if are_spatial_locations_preserved:
-                self.TotalPixelMatrixOriginSequence = \
-                    source_images[0].TotalPixelMatrixOriginSequence
-                self.TotalPixelMatrixRows = \
-                    source_images[0].TotalPixelMatrixRows
-                self.TotalPixelMatrixColumns = \
-                    source_images[0].TotalPixelMatrixColumns
+            )
+            if are_spatial_locations_preserved and is_tiled:
+                self.TotalPixelMatrixOriginSequence = deepcopy(
+                    src_img.TotalPixelMatrixOriginSequence
+                )
+                self.TotalPixelMatrixRows = src_img.TotalPixelMatrixRows
+                self.TotalPixelMatrixColumns = src_img.TotalPixelMatrixColumns
+            elif are_spatial_locations_preserved and ~is_tiled:
```

Comment:
Unless I'm missing something...

```suggestion
            elif are_spatial_locations_preserved and not is_tiled:
```